warehouse/forklift/legacy.py

```diff
@@ -260,85 +260,66 @@ def _is_valid_dist_file(filename, filetype):
     a valid distribution file.
     """
 
-    # If our file is a zipfile, then ensure that it's members are only
-    # compressed with supported compression methods.
-    if zipfile.is_zipfile(filename):
-        # Ensure the compression ratio is not absurd (decompression bomb)
-        compressed_size = os.stat(filename).st_size
-        with zipfile.ZipFile(filename) as zfp:
-            decompressed_size = sum(e.file_size for e in zfp.infolist())
-        if (
-            decompressed_size > COMPRESSION_RATIO_MIN_SIZE
-            and decompressed_size / compressed_size > COMPRESSION_RATIO_THRESHOLD
-        ):
-            sentry_sdk.capture_message(
-                f"File {filename} ({filetype}) exceeds compression ratio "
-                f"of {COMPRESSION_RATIO_THRESHOLD} "
-                f"({decompressed_size}/{compressed_size})"
-            )
-            return False
+    if filename.endswith((".zip", ".whl")):
```

Comment:
This change replaces the original `is_zipfile()` check prior to running other operations with a string comparison.
Would it make sense to restore it to bail early, is is that check "heavy" compared to the try/except we have here?

```suggestion
    if filename.endswith((".zip", ".whl")) and zipfile.is_zipfile(filename):
```

Reply:
There is a subtlety here: if only one of these conditionals is `False`, the section will be skipped. We need to check the value of `is_zipfile` separately and return if it is `False`: 3e3f0b2